src/pages/SignUpPage/SignUp.jsx

```diff
@@ -1,28 +1,165 @@
-import React from 'react';
+import React, { useReducer, useState } from 'react';
 import { Link } from 'react-router-dom';
 import "./SignUp.css";
+import signUpAnimation from "./../../../src/assets/images/SVG/signup.svg"
+import axios from 'axios';
+import { useNavigate as navigate } from 'react-router';
+import { useAuth } from '../../context/AuthContext';
+
+const SignUpDetails = (state,action) =>{
+    console.log(state,action);
+    console.log(action.email,action.firstName,action.lastName)
+    if(action.email){
+        return {...state,email : action.email}
+    }
+    if(action.firstName){
+        return {...state,firstName : action.firstName}
+    }
+    if(action.lastName){
+        return {...state,lastName : action.lastName}
+    }
+    return {...state}
+}
+
 
 function SignUpPage() {
+    const [passwordCheckError,setPasswordCheckError] = useState(false);
+    const [confirmPassword,setConfirmPassword] = useState("");
+    const [state,dispatch]= useReducer(SignUpDetails,{
+        email : "",
+        password : "",
+        firstName :"",
+        lastName:"",
+    });
+    const {login ,setlogin } = useAuth();
+
+
+    function HasAlphabets(letter) {
+        for (let i = 0; i < letter.length; i++) {
+          let char = letter[i];
+          if ((char >= "A" && char <= "Z") || (char >= "a" && char <= "z")) {
+            return true;
+          }
+        }
+      }
+      function HasNumber(letter) {
+        for (let i = 0; i < letter.length; i++) {
+          let char = letter[i];
+          if (!isNaN(char)) {
+            return true;
+          }
+        }
+      }
```

Comment:
I would suggest if you are using `regex` in your code than maintain everywhere in your code.

otherwise it is fine.

Reply:
for now, I will keep it this way and target on completing basic implementation once I am done and have time I will optimize the code to regex